app/src/main/kotlin/pl/gov/mc/protego/ui/registration/onboarding/OnboardingViewModel.kt

```diff
@@ -0,0 +1,85 @@
+package pl.gov.mc.protego.ui.registration.onboarding
+
+import androidx.annotation.DrawableRes
+import androidx.annotation.StringRes
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import pl.gov.mc.protego.R
+
+class OnboardingViewModel : ViewModel() {
+
+    private val pages = listOf(
+        PageInfo(
+            illustration = R.drawable.onboarding_01,
+            title = R.string.onboarding_hello_title,
+            description = R.string.onboarding_hello_description,
+            statusLegendVisible = false,
+            backButtonVisible = false,
+            nextButtonVisible = true
+        ),
+        PageInfo(
+            illustration = R.drawable.onboarding_02,
+            header = R.string.onboarding_status_title,
+            description = R.string.onboarding_status_description,
+            statusLegendVisible = true,
+            backButtonVisible = true,
+            nextButtonVisible = true
+        ),
+        PageInfo(
+            illustration = R.drawable.onboarding_03,
+            header = R.string.onboarding_bluetooth_title,
+            description = R.string.onboarding_bluetooth_description,
+            statusLegendVisible = false,
+            backButtonVisible = true,
+            nextButtonVisible = true
+        ),
+        PageInfo(
+            illustration = R.drawable.onboarding_04,
+            header = R.string.onboarding_sharing_title,
+            description = R.string.onboarding_sharing_description,
+            statusLegendVisible = false,
+            backButtonVisible = true,
+            nextButtonVisible = true
+        )
+    )
+
+    private var currentPageIndex = 0
+        set(value) {
+            field = value
+            page.value = pages[currentPageIndex]
+        }
+
+    val page = MutableLiveData<PageInfo>()
+    val navigateToRegistration = MutableLiveData<Unit>()
+    val finishApplication = MutableLiveData<Unit>()
```

Comment:
Wystawianie `MutableLiveData` w publicznym API ViewModeli nie jest dobrą praktyką. Sugeruję użycie https://kotlinlang.org/docs/reference/properties.html#backing-properties
w tym przypadku wyglądałoby to tak:

```
val page = LiveData<PageInfo>
    get() = _page

private val _page = MutableLiveData<PageInfo>()
```

Dzięki temu, wewnątrz ViewModelu publikujemy eventy do `_page`, a "świat zewnętrzny" może je tylko obserwować. Obecne rozwiazanie pozwala na to aby np. Activity publikowało eventy do tych obiektów LiveData.

Reply:
Masz rację, moje przeoczenie.